Describe the invalid precomputed address issue in the tools readme (#458)
hashgraph/hedera-json-rpc-relay: tools/README.md

Note: On contract deployment, most of the tools (e.g. [ethers](https://docs.ethers.io/v5/api/utils/address/#utils--contract-addresses)) pre-compute the contract address on the client-side, based
on sender address and nonce. In the Hedera ecosystem, it's not like that, where it's just the next available id.

> 

Reviewer: Update the Note:" to add
"[ethers js](https://docs.ethers.io/v5/api/utils/address/#utils--contract-addresses) and therefore Hardhat usage are impacted by this address calculation difference with the details captured [here](https://github.com/ethers-io/ethers.js/discussions/3141).
An extra step to retrieve the valid Hedera contract address is required to workaround this challenge, example workarounds are provided below"

Reviewer: You didn't add the link to the discussion item on the etherejs repo for the "here"

> #### Option 1

Reviewer: Please update the X mark in the table to a yellow caution/exclamation.

```typescript
// init the contract factory
const factory = new ethers.ContractFactory(contractJson.abi, contractJson.bytecode, wallet);
// deploy the contract
let contract = await factory.deploy();

> // wait till the transaction is mined and get the contract address from the receipt

Reviewer: ```suggestion
// wait till the transaction has reached consensus and get the contract address from the receipt
```

const { contractAddress } = await contract.deployTransaction.wait();

// re-init the contract with the deployed address
contract = new ethers.Contract(contractAddress, contractJson.abi, wallet);
```

#### Option 2
```typescript
// init the contract factory
const factory = new ethers.ContractFactory(contractJson.abi, contractJson.bytecode, wallet);
// deploy the contract
let contract = await factory.deploy();

> // wait for transaction to be mined

Reviewer: ```suggestion
// wait for transaction to reach consensus 
```
